src/app/views/authentication/Authentication.tsx

```diff
@@ -1,182 +1,152 @@
-          }
-        }
-        if (accessToken) {
-          try {
-            const query = {
-              sampleURL : USER_INFO_URL,
-            };
-            const userInfo = (queryActions) ? await queryActions.runQuery(query) : null;
-            const jsonUserInfo = userInfo.response.body;
-            authenticated.user = {...{},
-                                  displayName: jsonUserInfo.displayName,
-                                  emailAddress: jsonUserInfo.mail || jsonUserInfo.userPrincipalName,
-                                  profileImageUrl: '',
-            };
-            if (actions) {
-              actions.authenticateUser(authenticated);
-              this.setState({
-                authenticated,
-              });
-            }
-            try {
-              const pictureQuery = {
-                sampleURL : USER_PICTURE_URL,
-              };
-              const userPicture = (queryActions) ? await queryActions.runQuery(pictureQuery) : null;
-              const buffer = await userPicture.response.body.arrayBuffer();
-              const blob = new Blob([buffer], { type: 'image/jpeg' });
-              const imageUrl = URL.createObjectURL(blob);
-              if (actions) {
-                authenticated = this.state.authenticated;
-                authenticated.user.profileImageUrl = imageUrl;
-                actions.authenticateUser(authenticated);
-                this.setState({
-                  authenticated,
-                  loading: false,
-                });
-              }
-            } catch (e) {
-              if (actions) {
-                authenticated = this.state.authenticated;
-                authenticated.user.profileImageUrl = '';
-                actions.authenticateUser(authenticated);
-                this.setState({
-                  authenticated,
-                  loading: false,
-                });
-              }
-            }
-          } catch (e) {
-            // tslint:disable-next-line:no-console
-            console.log(e);
-          }
-        }
-      });
+  private async getUserInfo(queryActions: any) {
+    const userInfo = (queryActions) ? await queryActions.runQuery({
+      sampleURL: USER_INFO_URL,
+    }) : null;
+    const jsonUserInfo = userInfo.response.body;
+    return jsonUserInfo;
+  }
+
+  private async getImageUrl(queryActions: any) {
+    const userPicture = (queryActions) ? await queryActions.runQuery({
+      sampleURL: USER_PICTURE_URL,
+    }) : null;
+    const buffer = await userPicture.response.body.arrayBuffer();
+    const blob = new Blob([buffer], { type: 'image/jpeg' });
+    const imageUrl = URL.createObjectURL(blob);
+    return imageUrl;
   }
 
   public render() {
-    const { authenticated, loading } = this.state;
-    const buttonLabel = authenticated.status ? 'sign out' : 'sign in';
+    const { authenticatedUser, loading } = this.state;
+    const buttonLabel = authenticatedUser.status ? 'sign out' : 'sign in';
```

Comment:
Are we not concerned about i18n as of now ? (sign out and sign in)

Reply:
It's factored in in the component level. the words are passed in to the component and the component formats the strings in the required language